### ISSUE TITLE
Add option to turn off ordering/OrderedDict

### ISSUE THREAD

**Issue by sloria** [tomchristie/django-rest-framework#3445]
Using `OrderedDict` can results in a noticeable performance hit with large result sets. Furthermore, JSON is inherently unordered, so ordering by default seems a bit strange. In any case, it would be nice if ordering could be turned off, perhaps in `class Meta`.

**Comment by xordoquy** [comment]
@sloria do you have benchmarks we could run that supports the assertion that "Using OrderedDict can results in a noticeable performance hit with large result sets" ?

**Comment by aklim007** [comment]
Using OrderedDict can results in a noticeable performance hit with large result sets (Python <3.5)
https://docs.python.org/3.5/whatsnew/3.5.html#whatsnew-ordereddict

**Comment by xordoquy** [comment]
Interesting will run some benchs to figure how this impacts serializers

**Comment by xordoquy** [comment]
As it appears after a fast bench run under python 3.4 and 3.5, it looks like indeed we get an improvement using Python 3.5. With very basic tests - understand with a low test sample - it looks like it's around 10% faster for simple cases (1 User object) to almost twice as fast (serializing a list of 10,000 User objects).
More serious benchmark would make improvement clearer but it indeed seems there could be improvement with OrderedDict.

**Comment by xordoquy** [comment]
OK, so, I've did a very dirty trick to see how changing `OrderedDict` to `dict` affects bench (simply replacing `OrderedDict` by `dict` in the compat module).
My set of serializer benchs shows that there's a clear improvement.
However, I'm going to close this since the replacement is slightly faster than running the same thing with Python 3.5

TLDR: use Python 3.5 or pypy if you need improvements.

**Comment by xordoquy** [comment]
Running https://github.com/xordoquy/django-rest-framework-benchmark I got:

Regular (py34)
```
platform darwin -- Python 3.4.3, pytest-2.8.0, py-1.4.30, pluggy-0.3.1
rootdir: /Users/xordoquy/Documents/Devs/django-rest-framework-benchmark, inifile: tox.ini
plugins: benchmark-2.5.0, cov-2.1.0, django-2.8.0
collected 28 items 

tests/test_serializers.py ............................

----------- benchmark: 28 tests, min 5 rounds (of min 25.00us), 1.00s max time, timer: time.perf_counter -----------
Name (time in us)                                    Min           Max          Mean      StdDev  Rounds  Iterations
--------------------------------------------------------------------------------------------------------------------
test_model_serializer[1]                       1531.1270    21947.9590     1853.7991   1040.1013     409           1
test_model_serializer[10]                      2037.9960     4329.4490     2479.8290    371.7783     354           1
test_model_serializer[100]                     6981.3860    36568.0610     9184.4550   4702.2583     103           1
test_model_serializer[1000]                   62360.8090   125847.0910    81054.3544  23729.8744      11           1
test_model_serializer[10000]                 774252.8000   844168.0730   814342.4708  26451.8333       5           1
test_serializer[1]                              554.2970     1705.6580      714.0952    159.0823    1188           1
test_serializer[10]                             986.5690     3449.8750     1281.0813    281.2017     786           1
test_serializer[100]                           5203.5250   117233.7010     7811.6896   9864.1353     153           1
test_serializer[1000]                         52167.7720   116474.4110    69350.0179  22802.4677      20           1
test_serializer[10000]                       679456.0360   711286.9070   697165.3338  11510.5307       5           1
test_model_deserializer[1]                     1792.6190     3754.5960     2059.8204    234.5598     483           1
test_model_deserializer[10]                    2663.2260     5200.6750     3205.4533    476.2476     320           1
test_model_deserializer[100]                  11057.8160    16053.3200    12608.8597   1042.0666      74           1
test_model_deserializer[1000]                 99005.5650   152394.3800   111440.4570  22975.6452       5           1
test_model_deserializer[10000]              1148569.2010  1209121.9680  1178610.6498  21629.4032       5           1
test_deserializer[1]                            658.7750   100507.3180      935.1854   3202.0066     972           1
test_deserializer[10]                          1502.5080     3819.5370     1913.4068    337.2236     458           1
test_deserializer[100]                         9676.9160    38003.2790    11991.1393   4021.0939      81           1
test_deserializer[1000]                       96151.9790   151062.9960   111694.0268  22289.6087       8           1
test_deserializer[10000]                    1189377.8290  1282640.6520  1229667.3378  41343.1301       5           1
test_model_serializer_field_qt[1]               324.3240   119176.3940      477.4031   2997.5089    1572           1
test_model_serializer_field_qt[10]              904.8030     2365.8470     1174.0530    252.5083     662           1
test_model_serializer_field_qt[100]            6556.3430    39916.9410     8722.0814   3800.4830     127           1
test_model_serializer_field_qt[1000]          76313.9620   132059.1030    89827.6392  19719.1332      13           1
test_model_serializer_field_qt_10[1]            429.3070     1583.1000      529.9858    125.7419     973           1
test_model_serializer_field_qt_10[10]          1305.4420    31057.4820     1739.6421   1203.5205     652           1
test_model_serializer_field_qt_10[100]        10388.8120    45613.8900    13661.6349   5928.1684      79           1
test_model_serializer_field_qt_10[1000]      116420.4380   173322.7650   140352.0853  26247.5573       7           1
--------------------------------------------------------------------------------------------------------------------
```

Regular (py35)
```
platform darwin -- Python 3.5.0, pytest-2.8.0, py-1.4.30, pluggy-0.3.1
rootdir: /Users/xordoquy/Documents/Devs/django-rest-framework-benchmark, inifile: tox.ini
plugins: benchmark-2.5.0, cov-2.1.0, django-2.8.0
collected 28 items 

tests/test_serializers.py ............................

---------- benchmark: 28 tests, min 5 rounds (of min 25.00us), 1.00s max time, timer: time.perf_counter ---------
Name (time in us)                                   Min          Max         Mean      StdDev  Rounds  Iterations
-----------------------------------------------------------------------------------------------------------------
test_model_serializer[1]                      1323.6980    3311.7280    1575.5338    296.2319     464           1
test_model_serializer[10]                     1693.3150   22225.4000    2036.0893    987.1322     483           1
test_model_serializer[100]                    5333.3960   10996.0570    6309.4004    937.9992     160           1
test_model_serializer[1000]                  45065.4870   58078.5530   50076.1208   3093.6919      22           1
test_model_serializer[10000]                462491.3540  528767.0210  483065.2260  27077.5374       5           1
test_serializer[1]                             483.3640  148576.2460     890.2525   4261.5862    1210           1
test_serializer[10]                            787.7810    5008.8410    1474.3909    583.4401     615           1
test_serializer[100]                          3668.7180    8572.5560    4503.4436    791.9609     271           1
test_serializer[1000]                        34221.8070   50411.7060   39762.6479   3814.7521      27           1
test_serializer[10000]                      373814.9550  469074.7420  401389.1970  38350.2325       5           1
test_model_deserializer[1]                    1511.0090    4283.8230    1865.1757    374.3047     431           1
test_model_deserializer[10]                   2222.3320   73108.3010    2829.9259   3547.4396     401           1
test_model_deserializer[100]                  9003.9980   14772.9180   10551.3659   1247.9315      94           1
test_model_deserializer[1000]                79099.3370   93758.1190   85699.4920   3863.5251      13           1
test_model_deserializer[10000]              828494.9400  957208.1650  864424.1014  52602.0711       5           1
test_deserializer[1]                           583.4970   98729.7660     858.0967   2905.7453    1145           1
test_deserializer[10]                         1272.6510    2768.3060    1575.4474    306.6694     660           1
test_deserializer[100]                        7882.3190   13093.7550    9389.6254   1122.7821      98           1
test_deserializer[1000]                      78345.3570   93296.0860   86113.4913   4032.2973      13           1
test_deserializer[10000]                    819004.9410  925874.8780  845333.6168  45216.1445       5           1
test_model_serializer_field_qt[1]              241.6890    1432.3580     309.7612    104.9069    1955           1
test_model_serializer_field_qt[10]             736.1190    2651.2690     965.3380    240.0386     754           1
test_model_serializer_field_qt[100]           5458.1440   72392.8890    7730.2472   5755.6200     137           1
test_model_serializer_field_qt[1000]         61458.1260  127554.7000   75305.2709  21248.0019      15           1
test_model_serializer_field_qt_10[1]           302.2350   46592.0450     399.5003   1038.9710    2002           1
test_model_serializer_field_qt_10[10]         1013.4250    3063.0310    1306.3903    284.8935     930           1
test_model_serializer_field_qt_10[100]        8044.1040   16989.1190   10325.4516   1596.9947     108           1
test_model_serializer_field_qt_10[1000]      94992.7430  177172.9270  108710.9727  27926.2989       8           1
-----------------------------------------------------------------------------------------------------------------
```

dict solution (py34)
```
platform darwin -- Python 3.4.3, pytest-2.8.0, py-1.4.30, pluggy-0.3.1
rootdir: /Users/xordoquy/Documents/Devs/django-rest-framework-benchmark, inifile: tox.ini
plugins: benchmark-2.5.0, cov-2.1.0, django-2.8.0
collected 28 items 

tests/test_serializers.py ............................

---------- benchmark: 28 tests, min 5 rounds (of min 25.00us), 1.00s max time, timer: time.perf_counter ---------
Name (time in us)                                   Min          Max         Mean      StdDev  Rounds  Iterations
-----------------------------------------------------------------------------------------------------------------
test_model_serializer[1]                      1328.3960    4666.5560    1588.0945    307.7579     472           1
test_model_serializer[10]                     1692.2470   31043.0090    2111.9659   1429.1086     455           1
test_model_serializer[100]                    5216.5920   15302.6930    6042.2871    932.2768     165           1
test_model_serializer[1000]                  41824.6960   46309.2670   43529.7272   1504.7214      21           1
test_model_serializer[10000]                417680.5310  440698.3300  424140.8324   9841.2681       5           1
test_serializer[1]                             444.0910    1427.1660     569.6296    145.0418    1119           1
test_serializer[10]                            740.8130    1990.4450     925.3592    171.2647    1249           1
test_serializer[100]                          3563.5980    7716.1950    4291.2064    643.9478     242           1
test_serializer[1000]                        32398.2170   40011.3930   35685.2714   2261.5785      28           1
test_serializer[10000]                      335060.4990  345095.9840  340628.4206   3638.0391       5           1
test_model_deserializer[1]                    1501.3770    3855.1230    1774.9556    302.9319     471           1
test_model_deserializer[10]                   2163.9440    4597.3390    2549.8253    349.9334     395           1
test_model_deserializer[100]                  8324.7390   11642.3680    9415.8458    815.8533     110           1
test_model_deserializer[1000]                71212.6920   84797.0920   76934.6911   3825.3930      14           1
test_model_deserializer[10000]              729360.1750  768159.1140  748727.6256  15145.5764       5           1
test_deserializer[1]                           535.7540    2350.3240     680.1070    180.0355    1176           1
test_deserializer[10]                         1190.8210    3286.9040    1487.8237    264.7787     657           1
test_deserializer[100]                        7455.8390   10994.1790    8471.7772    789.1690     126           1
test_deserializer[1000]                      72267.3570   79099.1920   75384.9611   1957.0578      14           1
test_deserializer[10000]                    734897.7640  752700.0390  743519.6516   7946.3869       5           1
test_model_serializer_field_qt[1]              226.6260     892.9200     270.3384     75.9938    1680           1
test_model_serializer_field_qt[10]             707.7960    4206.6080     909.6734    256.7823     944           1
test_model_serializer_field_qt[100]           5409.0380  142359.1690    7698.0273  10592.9460     166           1
test_model_serializer_field_qt[1000]         62240.9900  130451.9510   72030.0161  19588.0517      11           1
test_model_serializer_field_qt_10[1]           285.1050   43623.3580     365.2023   1137.3346    1456           1
test_model_serializer_field_qt_10[10]          970.6570    2908.6300    1220.1127    228.4380     932           1
test_model_serializer_field_qt_10[100]        8402.7200   13084.5230    9735.2830   1140.2417      98           1
test_model_serializer_field_qt_10[1000]      87035.0150  155635.4350  101169.1456  22367.7691       8           1
-----------------------------------------------------------------------------------------------------------------
```
